File: questModule/src/main/java/ru/nekit/android/qls/quest/types/TextQuest.java
package ru.nekit.android.qls.quest.types;

import android.text.InputType;

import ru.nekit.android.qls.quest.QuestionType;
import ru.nekit.android.qls.quest.base.AbstractQuest;

public class TextQuest extends AbstractQuest {

    public String[] questionStringArray;
    private String[] answerStringArray;

    public TextQuest(QuestionType questionType) {
        setQuestionType(questionType);
    }

    @Override
    public Object getAnswer() {
        return getAnswerString();
    }

    public void setAnswer(String answer) {
        answerStringArray = new String[]{answer};
    }

    public String getQuestionString() {
        StringBuilder questionString = new StringBuilder();
        int answerItemPosition = 0;
        for (String questionItem : questionStringArray) {
            if ("".equals(questionItem)) {
                questionString.append(answerStringArray[answerItemPosition]);
                answerItemPosition++;
            } else {
                questionString.append(questionItem);
            }
        }
        return questionString.toString();
    }

    private String getAnswerString() {
        StringBuilder answerString = new StringBuilder();
        for (String answerItem : answerStringArray) {
            answerString.append(answerItem);
        }
        return answerString.toString();
    }

    @Override
    public Class getAnswerClass() {
        return String.class;
    }

    @Override
    public int getAnswerInputType() {
        return InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS;
    }
}

File: questPresentation/src/main/java/ru/nekit/android/qls/quest/types/TextQuest.kt
package ru.nekit.android.qls.quest.types

import ru.nekit.android.qls.domain.model.quest.Quest

class TextQuest : Quest() {

    var questionStringArray: Array<String>? = null
    private var answerStringArray: Array<String>? = null

    override val answer: String
        get() = answerString

    val questionString: String
        get() {
            val questionString = StringBuilder()
            var answerItemPosition = 0
            for (questionItem in questionStringArray!!) {
                if ("" == questionItem) {
                    questionString.append(answerStringArray!![answerItemPosition])
                    answerItemPosition++
                } else {
                    questionString.append(questionItem)
                }
            }
            return questionString.toString()
        }

    private val answerString: String
        get() {
            val answerString = StringBuilder()
            for (answerItem in answerStringArray!!) {
                answerString.append(answerItem)
            }
            return answerString.toString()
        }

    override val answerClass: Class<*>
        get() = String::class.java

    fun setAnswer(answer: String) {
        answerStringArray = arrayOf(answer)
    }

}